### PR TITLE
Convert to `docker buildx bake`, drop unsupported windows images

### PR DESCRIPTION
**Pull request by timja**
Ported https://github.com/jenkinsci/docker with some test adaptions

Relates to https://github.com/jenkinsci/docker-agent/issues/109

cc @MarkEWaite @dduportal

### CODE REVIEW

**Comment by timja** [comment]
4m => 1m for build

File: Makefile [hunk 1]
```diff
@@ -1,73 +1,90 @@
...
-test-alpine: test-run-alpine
+list: check-reqs
+	@set -x; make --silent show | jq -r '.target | path(.. | select(.platforms[] | contains("linux/$(ARCH)"))?) | add'
 
-test-archlinux: FOLDER=11/archlinux
-test-archlinux: test-run-archlinux
+bats:
+	git clone https://github.com/bats-core/bats-core bats ;\
+	cd bats ;\
+	git checkout eac1e9d047b2b8137d85307fc94439c90bdc25ae
```
Comment (dduportal) [inline Makefile:57]: @timja I forgot why did we use this particular commit? Was it the "latest known to be working" at this time?

Comment (timja) [reply Makefile:57]: yeah, there was some parallel fixes after the last released version

Comment (dduportal) [reply Makefile:57]: Do you mind if we fix it to the tag `1.4.1` (instead of a commit) ?

Comment (dduportal) [reply Makefile]: (I've pushed https://github.com/jenkinsci/docker-agent/pull/182/commits/a2c68d617b4bc2bde902d90fb1ea513506f3596a as an example: feel free to remove it or ask me to remove it)

Comment (timja) [reply Makefile]: 1.4.0 has the fixes by the looks of it so this is good thanks!

**Pull request merged**